plugins/doc_fragments/api.py

```diff
@@ -95,3 +95,35 @@ class ModuleDocFragment(object):
   - ref: ansible_collections.community.routeros.docsite.api-guide
     description: How to connect to RouterOS devices with the RouterOS API
 '''
+
+    RESTRICT = r'''
+options:
+  restrict:
+    type: list
+    elements: dict
+    suboptions:
+      field:
+        description:
+          - The field whose values to restrict.
+        required: true
+        type: str
+      values:
+        description:
+          - The values of the field to limit to.
+          - >-
+            Note that the types of the values are important. If you provide a string V("0"),
+            and librouteros converts the value returned by the API to the integer V(0),
+            then this will not match. If you are not sure, better include both variants:
+            both the string and the integer.
+          - Use V(none) for disabled values.
+          - Either O(restrict[].values) or O(restrict[].regex), but not both, must be specified.
+        type: list
+        elements: raw
+      regex:
+        description:
+          - A regular expression matching values of the field to limit to.
+          - Note that all values will be converted to strings before matching.
+          - It is not possible to match disabled values with regular expressions.
```

Comment:
With an additional option, e.g. `negative`, matching disabled/non-existent values should be possible. Consider the following config:

```
/ip dns static
add address=1.2.3.4 comment="# managed, lease=foobar123" disabled=no name=test.example.com ttl=1d
add address=2.3.4.5 disabled=no name=static.example.com ttl=1d
```

If I understand the logic correctly it's not possible to restrict the module's operations to the entries _without_ comment.

Reply:
While inversion/negation makes sense, it is possible to restrict to everything without comments by specifying:
```
  - field: ...
    values:
      - ~
```

Reply:
You're right. My regex-colored glasses made me miss `values: [null]`. That considered it might be more flexible to support both matchers at the same time:

```yaml
restrict:
- value: "hello"
- value: null
- regex: "^foo"
- value: "world"
  exclude: true
- regex: "^# magic"
  exclude: true
```

Reply:
I changed the behavior in 6e59f36a1b3871b6262d5c8638faa4547f758b67, now you have two additional options `match_disabled` and `invert`, and `values` and `regex` are no longer mutually exclusive. So you can do:
```yaml
restrict:
  # Match all entries which do not have a comment
  - field: comment
    match_disabled: true
    invert: true
  # Match chain "input" and all chains starting with "foo"
  - field: chain
    values:
      - input
    regex: "^foo"
```

Reply:
LGTM, thank you a lot!